joinmarket/blockchaininterface.py

```diff
@@ -633,12 +658,26 @@ def sync_addresses(self, wallet):
                             unused_addr_count = 0
                         else:
                             unused_addr_count += 1
-
-                if last_used_addr == '':
-                    wallet.index[mix_depth][forchange] = 0
-                else:
-		    next_avail_idx = max([wallet.addr_cache[last_used_addr][2]+1,
-		                        wallet.index_cache[mix_depth][forchange]])
+		#index setting here depends on whether we broke out of the loop
+		#early; if we did, it means we need to prepare the index
+		#at the level of the last used address or zero so as to not
+		#miss any imports in add_watchonly_addresses.
+		#If we didn't, we need to respect the index_cache to avoid
+		#potential address reuse.
+		if breakloop:
+		    if last_used_addr == '':
```

Comment:
This "if breakloop" section is as before #574 . we will always start the addr_req_count*3 final step from either zero or the last used address; in the latter case, since no address could be marked "last used" without already having been imported, I believe it's impossible to end up with a "gap" of missed imports.